src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorProjectService.cs

```diff
@@ -139,41 +139,90 @@ await AddOrUpdateProjectCoreAsync(
             .ConfigureAwait(false);
     }
 
+    public async Task AddDocumentsToMiscProjectAsync(ImmutableArray<string> filePaths, CancellationToken cancellationToken)
+    {
+        await WaitForInitializationAsync().ConfigureAwait(false);
+
+        await _projectManager
+            .UpdateAsync(
+                (updater, cancellationToken) =>
+                {
+                    var projects = _projectManager.GetProjects();
+
+                    // For each file, check to see if it's already in a project.
+                    // If it is, we don't want to add it to the misc project.
+                    foreach (var filePath in filePaths)
+                    {
+                        foreach (var project in projects)
+                        {
+                            if (project.ContainsDocument(filePath))
+                            {
+                                if (cancellationToken.IsCancellationRequested)
+                                {
+                                    return;
+                                }
+
+                                AddDocumentToMiscProjectCore(updater, filePath);
+                            }
```

Comment:
I just noticed that this logic does exactly the opposite of what it's supposed to do. A fix is incoming.